### PR TITLE
Task2

### PR DESCRIPTION
**Pull request by LiuIhor**
It is my second project Simple Search Engine

### CODE REVIEW

File: src/main/java/task2/Main.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        List<String> dataFromFile = new ArrayList<>();
+        Map<String, Set<Integer>> invertedIndex;
+        if(args[0].equals("--data")) {
+            File file = new File(args[1]);
+
+            try (Scanner scanner = new Scanner(file)) {
+                while (scanner.hasNext()) {
+                    dataFromFile.add(scanner.nextLine()); // fill the list with data lines from file
+                }
+                scanner.close();
```
Comment (romaro488) [inline src/main/java/task2/Main.java]: Please remove this line. Scanner will be closed automatically in the line 17

File: src/main/java/task2/Controller.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        for (String person : dataFromFile) {
+            System.out.println(person);
+        }
+    }
+
+    /**
+     * Method to display all founded data
+     * */
+    private static void printResult(Set<Integer> searchResult, List<String> dataFromFile) {
+        if (searchResult.size() == 0) {
```
Comment (romaro488) [inline src/main/java/task2/Controller.java]: You can use isEmpty() method

File: src/main/java/task2/ContextSearch.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    }
+
+    ContextSearch() { }
+
+    public void setSearchingMethod(StrategySearch searchingMethod) {
+        this.searchingMethod = searchingMethod;
+    }
+
+    public Set<Integer> search(Map<String, Set<Integer>> invertedIndex, List<String> dataFromFile, String query) {
+        return this.searchingMethod.search(invertedIndex, dataFromFile, query);
```
Comment (romaro488) [inline src/main/java/task2/ContextSearch.java]: why do you need `this`?

File: src/main/java/task2/Controller.java [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+    private static Set<Integer>  strategySearch(Map<String, Set<Integer>> invertedIndex, List<String> dataFromFile) {
+        Scanner sc = new Scanner(System.in);
+        System.out.println("Select a matching strategy: ALL, ANY, NONE");
+        String searchingMethod = sc.nextLine();
+        System.out.println("Enter a name or email to search all suitable people.");
+        String query = sc.nextLine();
+        ContextSearch contextSearch = new ContextSearch();
+        switch (searchingMethod) {
+            case "ALL":
```
Comment (romaro488) [inline src/main/java/task2/Controller.java]: Please use enum for this values

File: src/main/java/task2/StrategySearchAny.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+ * */
+public class StrategySearchAny implements StrategySearch{
+
+    @Override
+    public Set<Integer> search(Map<String, Set<Integer>> invertedIndex, List<String> dataFromFile, String query) {
+        Set<String> querySet = Arrays.stream(query.trim().split(" ")).collect(Collectors.toSet());
+        Set<Integer> searchResult = new HashSet<>();
+
+        for (String q : querySet) {
+            Set<Integer> values = invertedIndex.getOrDefault(q.toUpperCase(Locale.ROOT), new HashSet<>(Set.of()));
```
Comment (romaro488) [inline src/main/java/task2/StrategySearchAny.java]: Why do you need `Set.of()` in the initialising  new HashSet<>?

File: src/main/java/task2/StrategySearchAll.java [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+/**
+ * It is implementation of the interface with the implemented searching method
+ * */
+public class StrategySearchAll implements StrategySearch {
+
+    @Override
+    public Set<Integer> search(Map<String, Set<Integer>> invertedIndex, List<String> dataFromFile, String query) {
+        Set<String> querySet = Arrays.stream(query.trim().split(" ")).collect(Collectors.toSet());
+        Set<String> tempLine;
```
Comment (romaro488) [inline src/main/java/task2/StrategySearchAll.java]: What the reason create this Set here?

File: src/main/java/task2/Main.java (hunk 1 above)
Comment (LiuIhor) [reply src/main/java/task2/Main.java]: Done. Thanks

File: src/main/java/task2/Controller.java (hunk 2 above)
Comment (LiuIhor) [reply src/main/java/task2/Controller.java]: Done. Thanks.

File: src/main/java/task2/ContextSearch.java (hunk 3 above)
Comment (LiuIhor) [reply src/main/java/task2/ContextSearch.java]: "this" was deleted

File: src/main/java/task2/Controller.java (hunk 4 above)
Comment (LiuIhor) [reply src/main/java/task2/Controller.java]: Done.

File: src/main/java/task2/StrategySearchAny.java (hunk 5 above)
Comment (LiuIhor) [reply src/main/java/task2/StrategySearchAny.java]: Thanks. I removed Set.of()

File: src/main/java/task2/StrategySearchAll.java (hunk 6 above)
Comment (LiuIhor) [reply src/main/java/task2/StrategySearchAll.java]: Update.

**Pull request merged**